src/sinks/util/metadata.rs

```diff
@@ -0,0 +1,44 @@
+use super::request_builder::EncodeResult;
+
+/// Metadata for batch requests.
+#[derive(Clone, Debug)]
+pub struct BatchRequestMetadata {
+    /// Number of events represented by this batch request.
+    pub event_count: usize,
+    /// Size, in bytes, of the in-memory representation of all events in this batch request.
+    pub event_byte_size: usize,
+    /// Uncompressed size, in bytes, of the encoded events in this batch request.
+    pub encoded_uncompressed_size: usize,
+    /// Compressed size, in bytes, of the encoded events in this batch request, if compression was performed.
+    pub encoded_compressed_size: Option<usize>,
```

Comment:
Hmmm, perhaps? I have an idea of how to try and make this a little more builder-y, to avoid the chance for incorrect values to get stored... so let me give that a whirl. 👍🏻

Reply:
I'e significantly reworked `RequestMetadata` in 6b8a34b9f such that it must be built from something that is `EventCount + ByteSizeOf` to get the event count/event byte size, and then either given an `&EncodeResult` or a single `NonZeroUsize` for both encoded/wire size.

Now there is no way to pass in raw `usize` values that could be incorrectly set to zero, although it doesn't stop the source of those values from being zero... although those values could propagate to other places and cause issues, so I'm not sure `RequestMetadata` is the place to make a stand against zero values in general.